enderio-machines/src/main/java/com/enderio/machines/common/recipe/MachineRecipe.java

```diff
@@ -0,0 +1,46 @@
+package com.enderio.machines.common.recipe;
+
+import com.enderio.base.EnderIO;
+import com.enderio.base.common.recipe.DataGenSerializer;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.Container;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Recipe;
+import net.minecraftforge.common.crafting.CraftingHelper;
+import net.minecraftforge.common.crafting.conditions.ICondition;
+import net.minecraftforge.common.crafting.conditions.ModLoadedCondition;
+import net.minecraftforge.fluids.FluidStack;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public abstract class MachineRecipe<T extends MachineRecipe<T, C>, C extends Container> implements Recipe<C> {
+
+    @Override
+    public abstract DataGenSerializer<T, C> getSerializer();
+
+    protected Stream<ItemStack> getItemsInRecipe() {
+        return Stream.empty();
+    }
+
+    protected Stream<FluidStack> getFluidsInRecipe() {
+        return Stream.empty();
+    }
+
+    protected Stream<ResourceLocation> getOtherResourceLocations() {
+        return Stream.empty();
+    }
+
+    public final List<String> getModDependencies() {
+        return Stream.concat(Stream.concat(getItemsInRecipe().map(stack -> stack.getItem().getRegistryName()),
+            getFluidsInRecipe().map(fluidStack -> fluidStack.getFluid().getRegistryName())),
+            getOtherResourceLocations()) //merge all resourcelocations for this recipe
+            .map(ResourceLocation::getNamespace) //get their origin
+            .filter(string -> !string.equals("minecraft")) //remove minecraft, forge and enderio
+            .filter(string -> !string.equals("forge"))
+            .filter(string -> !string.equals(EnderIO.DOMAIN))
```

Comment:
Might want to add the other modids for the enderio modules after the registration change.

Reply:
Actually: I don't, because otherwise decoration could be an output, even though decoration might not be present.
But I will add enderio_machines

Reply:
perfect, thanks